src/unitxt/operators.py

```diff
@@ -508,75 +508,121 @@ def process_value(self, value: Any) -> Any:
+                continue
+            if isinstance(arg, dict):
+                for k, v in arg.items():
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should map strings, whereas key {k!s} is of type {type(k)}"
+                    assert isinstance(
+                        v, int
+                    ), f"{arg_name} should map to ints, whereas value {v!s} is of type {type(v)}"
+            else:
+                for k in arg:
+                    assert isinstance(
+                        k, str
+                    ), f"{arg_name} should be a list of strings, whereas member {k!s} is of type {type(k)}"
         super().verify()
 
     def prepare(self):
-        self.pats = (
-            self.suffixes
-            if isinstance(self.suffixes, list)
-            else [k for k, v in self.suffixes.items()]
-        )
-        total_weight = (
-            len(self.pats)
-            if isinstance(self.suffixes, list)
-            else sum([v for k, v in self.suffixes.items()])
-        )
-        self.weights = (
-            [1.0 / total_weight] * len(self.pats)
-            if isinstance(self.suffixes, list)
-            else [float(self.suffixes[p]) / total_weight for p in self.pats]
-        )
+        # Being an artifact, prepare is invoked before verify. Here we need verify before the actions
+        self.verify()
+        self.pats = {"prefixes": None, "suffixes": None}
+        self.weights = {"prefixes": None, "suffixes": None}
+
+        for end, end_name in zip(
+            [self.prefixes, self.suffixes], ["prefixes", "suffixes"]
+        ):
+            if end is None:
+                continue
+            self.pats[end_name] = (
+                end if isinstance(end, list) else [k for k, v in end.items()]
+            )
+            total_weight = (
+                len(self.pats[end_name])
+                if isinstance(end, list)
+                else sum([v for k, v in end.items()])
+            )
+            self.weights[end_name] = (
+                [1.0 / total_weight] * len(self.pats[end_name])
+                if isinstance(end, list)
+                else [float(end[p]) / total_weight for p in self.pats[end_name]]
+            )
 
         super().prepare()
 
     def process_value(self, value: Any) -> Any:
         assert value is not None, "input value should not be None"
         new_value = str(value)
-        if self.remove_existing_trailing_whitespaces:
-            new_value = new_value.rstrip()
-        new_value += get_random().choices(self.pats, self.weights, k=1)[0]
+        if self.remove_existing_whitespaces:
+            new_value = new_value.strip()
+        prefix = ""
+        suffix = ""
+        if self.prefixes:
+            prefix = "".join(
+                get_random().choices(
+                    self.pats["prefixes"], self.weights["prefixes"], k=1
+                )
+                * get_random().randint(1, 5)
```

Comment:
Let's have the number of repetition also be a parameter of the augmentor.

Reply:
OK. Done.